### PR TITLE
Interactivity improvements

### PR DESCRIPTION
**Pull request by ElDeveloper**
The new functionality comes from talking to @tanaes. When visualizing concentrations in a plate (as represented by a heatmap), it is useful to change the range of the colorscale, in order to do this, we've added a slider to manipulate this value, and set some appropriate defaults depending on the plate types. This will also be useful as we reuse this component for other steps in the processing pipeline.

Demo of the new functionality:
slidere

I've also cleaned up some of the JS code to avoid multiple traversals of the main arrays.

### CODE REVIEW

File: labman/gui/templates/library_pooling.html [hunk 1]
```diff
@@ -193,21 +221,59 @@
...
   };
 
+  /**
+   * Determine an appropriate clipping value for the heatmap's maximum value.
+   *
+   * @param {Float} plateType String describing the plate type.
+   * @returns A maximum expected value for the colorscale of a heatmap.
+   *
+   */
+  function clippingForPlateType(plateType) {
```
Comment (ElDeveloper) [inline labman/gui/templates/library_pooling.html]: @tanaes I am sure 💯 % that I got these wrong (forgot to write the numbers down), I would appreciate if you can remind me the correct values.

**Comment by josenavas** [comment]
Code looks good 👍

File: labman/gui/templates/library_pooling.html [hunk 2]
```diff
@@ -193,21 +221,59 @@
...
+    if (plateType == '16S library prep') {
+      return 100;
+    }
+    else {
```
Comment (tanaes) [inline labman/gui/templates/library_pooling.html]: maybe make this an `elif (plateType == 'shotgun library prep')` statement to explicitly address the possibility of other plate types?

These are pooling values, so correct defaults are 1000 for shotgun and 15 for 16S. 

IF using slider to clip, do we also need to set the slider range values?

File: labman/gui/templates/library_pooling.html [hunk 3]
```diff
@@ -193,21 +221,59 @@
...
       yaxis: {title: "Number Samples"},
       title: 'DNA Concentration Distribution'
     };
-    var histdata = [trace, blank_trace];
+    var histogramData = [nonBlankData, blankData];
+
+    Plotly.plot($heatmap.attr('id'), heatmapData, heatmapLayout);
+    Plotly.plot($histogram.attr('id'), histogramData, histogramLayout);
+
+    // setup some interactivity to clip the values in the histogram
```
Comment (tanaes) [inline labman/gui/templates/library_pooling.html]: Does this want to be simplified into a text box? The advantage is that it wouldn't require any default max values.

File: labman/gui/templates/library_pooling.html [hunk 4]
```diff
@@ -75,15 +78,41 @@
...
+   * @param {Float} defaultClipping A number representing the default clipping
+   * value for the heatmap. Depends on the processing stage and the type of
+   * data being processed.
    *
    */
-  function createHeatmap(plateId, concentrations, blanks, names) {
-    var container = $('#pool-results-' + plateId);
-    var heatmap = $('<div id="heatmap-' + plateId + '"></div>');
-    var histogram = $('<div id="histogram-' + plateId + '"></div>');
+  function createHeatmap(plateId, concentrations, blanks, names,
```
Comment (tanaes) [inline labman/gui/templates/library_pooling.html]: We want to also have this code in the quantify interface. 

Should we be locating this function in a different, centralized file and importing?

**Comment by ElDeveloper** [comment]
I've addressed the changes, mainly changing the slider for a spinner:

spinner

**Comment by tanaes** [comment]
niiice

**Pull request merged**